src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubRDDIterator.java

```diff
@@ -119,11 +122,14 @@ private SubscriberStub buildSubscriberClient() throws IOException {
     if (credentials != null) {
       builder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
     }
+    builder.setTransportChannelProvider(
+      SubscriberStubSettings.defaultGrpcTransportProviderBuilder()
+        .setMaxInboundMessageSize(MAX_MESSAGE_SIZE).build());
     builder.getSubscriptionSettings().setRetrySettings(PubSubSubscriberUtil.getRetrySettings());
     return GrpcSubscriberStub.create(builder.build());
   }
 
-  private List<ReceivedMessage> fetchAndAck() throws IOException {
+  private List<ReceivedMessage> fetchAndAck(long timeout) throws IOException {
```

Comment:
nit: the other name `expireTimeMillis` is better, timeout sounds like a duration and not a timestamp